snapcraft/plugins/rust.py

```diff
@@ -105,9 +106,18 @@ def _build_env(self):
         env = os.environ.copy()
         env.update({"RUSTC": self._rustc,
                     "RUSTDOC": self._rustdoc,
-                    "RUST_PATH": self._rustlib})
+                    "RUST_PATH": self._rustlib,
+                    'RUSTFLAGS': self._rustflags()})
         return env
 
+    def _rustflags(self):
+        ldflags = shell_utils.getenv('LDFLAGS')
+        rustldflags = ''
+        flags = {flag for flag in ldflags.split(' ') if flag}
```

Comment:
the `if flag` in the original fixes that, we can add `flag.strip() for flag in ...`  but I don't think we need that.